### PR TITLE
Use github code scanning instead of LGTM

### PR DESCRIPTION
**Pull request by liushilongbuaa**
**- What I did**
LGTM is deprecated in December 2022. Enable github code scanning instead.
result is pasted in comment.
**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by liushilongbuaa** [comment]
image

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# For more infomation, please visit: https://github.com/github/codeql-action
+
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+      - 'master'
+      - '202[0-9][0-9][0-9]'
```
Comment (SuvarnaMeenakshi) [inline .github/workflows/codeql-analysis.yml:9]: Is this change required for 2019/2018 branches?

**Review by xumia — APPROVED** [review]
